Modules/Core/Sources/Core/Stream/Stream.swift

```diff
@@ -1,52 +1,61 @@
 public enum StreamError : Error {
-    case closedStream(data: Data)
-    case timeout(data: Data)
+    case closedStream(buffer: Buffer)
+    case timeout(buffer: Buffer)
 }
 
 public protocol InputStream {
     var closed: Bool { get }
     func close()
-    func read(into buffer: inout Data, length: Int, deadline: Double) throws -> Int
+    
+    func read(into: UnsafeMutableBufferPointer<UInt8>, deadline: Double) throws -> Int
+    func read(upTo: Int, deadline: Double) throws -> Buffer
 }
 
 extension InputStream {
-    public func read(into buffer: inout Data, length: Int) throws -> Int {
-        return try read(into: &buffer, length: length, deadline: .never)
+    public func read(into: UnsafeMutableBufferPointer<UInt8>) throws -> Int {
+        return try read(into: into, deadline: .never)
     }
-
-    public func read(into buffer: inout Data, deadline: Double = .never) throws -> Int {
-        return try read(into: &buffer, length: buffer.count, deadline: deadline)
+    
+    public func read(upTo count: Int, deadline: Double = .never) throws -> Buffer {
+        return try Buffer(capacity: count) { try read(into: $0, deadline: deadline) }
     }
 }
 
 public protocol OutputStream {
     var closed: Bool { get }
     func close()
-    @discardableResult
-    func write(_ buffer: Data, length: Int, deadline: Double) throws -> Int
+    
+    func write(from: UnsafeBufferPointer<UInt8>, deadline: Double) throws
```

Comment:
I think it may be better to keep this unlabeled (`write(_:)` instead of `write(from:)`